### ISSUE TITLE
or filter

### ISSUE THREAD

**Issue by jotpe** [drolbr/Overpass-API#173]
Searching one object by ORed filter is done right now with:

``` c
(
 rel[wikidata=Q365];
 rel[~"^wikipedia(|:de)$"~"(|de:)Köln"];
)->.koeln;
.koeln out geom;
```

But if the object was already found by the first ky-pair, the second wikipedia filter will find the same relation again. The .koeln set removes double entries, so my result is fine.

Something like next example reduces useless queries. The || operator acts like everywhere else. If the first expression is true (means something was found), the rest will be skipped.

``` c
rel[wikidata=Q365] || [~"^wikipedia(|:de)$"~"(|de:)Köln"];
out geom;
```

What do you think?

**Comment by mmd-osm** [comment]
That's somewhat similar to what I proposed here already: https://github.com/drolbr/Overpass-API/issues/112#issuecomment-47277121

**Comment by jotpe** [comment]
Now I remember. But in this case it deals with different filter types (plain, regexp) and different filter keys. At first I tried the cheapest operation: looking for the exact wikidata tag. Followed by more complex filters, if nothing was found. And the || should skip following filters, when something was found. A single | could skip the skip.

**Comment by drolbr** [comment]
Implemented in commit 7bf3304e6995c5aecb5128b2cca1ca45e745afbc.
